lib/merkDb/MerkDbTransaction.js

```diff
@@ -0,0 +1,75 @@
+const MerkDbWrapper = require('./MerkDbTransactionWrapper');
+
+const MerkDBTransactionIsNotStartedError = require('./errors/MerkDBTransactionIsNotStartedError');
+const MerkDBTransactionIsAlreadyStartedError = require('./errors/MerkDBTransactionIsAlreadyStartedError');
+
+class MerkDbTransaction {
+  /**
+   *
+   * @param {Merk} merkDB
+   */
+  constructor(merkDB) {
+    this.merkDB = merkDB;
+    this.db = null;
+  }
+
+  /**
+   * Start new transaction in merk DB
+   *
+   * @return {MerkDbTransaction}
+   */
+  start() {
+    if (this.db) {
+      throw new MerkDBTransactionIsAlreadyStartedError();
+    }
+
+    this.db = new MerkDbWrapper(this.merkDB);
```

Comment:
Let’s call it `MerkDbInMemoryDecorator`?

Reply:
Why "InMemory"?

Reply:
Coz we store data in memory